lib/internal/test_runner/test.js

```diff
@@ -77,14 +76,23 @@ let kResistStopPropagation;
 
 function stopTest(timeout, signal) {
   if (timeout === kDefaultTimeout) {
-    return once(signal, 'abort');
+    const deferred = createDeferredPromise();
+
+    signal.addEventListener('abort', deferred.resolve, { __proto__: null, once: true });
+    deferred.cleanup = () => signal.removeEventListener('abort', deferred.resolve);
+    return deferred;
   }
-  return PromisePrototypeThen(setTimeout(timeout, null, { ref: false, signal }), () => {
-    throw new ERR_TEST_FAILURE(
-      `test timed out after ${timeout}ms`,
-      kTestTimeoutFailure,
-    );
-  });
+
+  return {
+    __proto__: null,
+    promise: PromisePrototypeThen(setTimeout(timeout, null, { ref: false, signal }), () => {
+      throw new ERR_TEST_FAILURE(
+        `test timed out after ${timeout}ms`,
+        kTestTimeoutFailure,
+      );
+    }),
+    cleanup: noop,
+  };
```

Comment:
There is a warning here too when using timeout on the hook but it will be in a different PR as I think there is a problem in the setTimeout rather than here

Reply:
Done in this PR in the end